cdap-ui/app/cdap/components/Replicator/Create/Content/Summary/ActionButtons/index.tsx

```diff
@@ -0,0 +1,164 @@
+import Button from '@material-ui/core/Button';
+import LoadingSVG from 'components/LoadingSVG';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { MyAppApi } from 'api/app';
+import { MyReplicatorApi } from 'api/replicator';
+import { Redirect } from 'react-router-dom';
+import If from 'components/If';
+
+const styles = (theme): StyleRules => {
+  return {
+    root: {
+      marginTop: '50px',
+      borderTop: `1px solid ${theme.palette.grey[300]}`,
+      paddingTop: '25px',
+      '& button': {
+        marginRight: '50px',
+      },
+    },
+    error: {
+      marginTop: '50px',
+      color: theme.palette.red[100],
+    },
+  };
+};
+
+enum REDIRECT_TARGET {
+  detail = 'detail',
+  list = 'list',
+}
+
+const ActionButtonsView: React.FC<ICreateContext & WithStyles<typeof styles>> = ({
+  classes,
+  getReplicatorConfig,
+  parentArtifact,
+  draftId,
+  saveDraft,
+  name,
+}) => {
+  const [deployLoading, setDeployLoading] = React.useState(false);
+  const [error, setError] = React.useState(null);
+  const [redirect, setRedirect] = React.useState(null);
+
+  function constructJson() {
+    const config = getReplicatorConfig();
+
+    return {
+      name,
+      artifact: {
+        ...parentArtifact,
+      },
+      config,
+    };
+  }
+
+  function deployReplicator() {
+    setDeployLoading(true);
+    const spec = constructJson();
+
+    const params = {
+      namespace: getCurrentNamespace(),
+      appName: spec.name,
+    };
+
+    MyAppApi.list({ namespace: getCurrentNamespace() }).subscribe(
+      (apps) => {
+        const existingAppName = apps.filter((app) => app.name === name);
+
+        if (existingAppName.length > 0) {
+          setError(`There is already an existing application "${name}"`);
+          setDeployLoading(false);
+          return;
+        }
+
+        MyReplicatorApi.publish(params, spec).subscribe(
+          () => {
+            MyReplicatorApi.deleteDraft({
+              namespace: getCurrentNamespace(),
+              draftId,
+            }).subscribe(null, null, () => {
+              setRedirect(REDIRECT_TARGET.detail);
+            });
+          },
+          (err) => {
+            setError(err);
+            setDeployLoading(false);
+          }
+        );
+      },
+      (err) => {
+        setError(err);
+        setDeployLoading(false);
+      }
+    );
+  }
+
+  function saveAndClose() {
+    saveDraft().subscribe(
+      () => {
+        setRedirect(REDIRECT_TARGET.list);
+      },
+      (err) => {
+        setError(err);
+      }
+    );
+  }
+
+  if (redirect) {
+    let redirectLink = `/ns/${getCurrentNamespace()}/replicator`;
+    if (redirect === REDIRECT_TARGET.detail) {
+      redirectLink = `${redirectLink}/detail/${name}`;
+    }
+
+    return <Redirect to={redirectLink} />;
+  }
+
+  return (
+    <React.Fragment>
+      <If condition={error}>
+        <div className={classes.error}>{JSON.stringify(error, null, 2)}</div>
```

Comment:
nit: We might be good if we just display the message from backend. Is there a possibility for us to show stack trace or a JSON here?